packages/devtools_app/lib/src/screens/memory/framework/memory_controller.dart

```diff
@@ -85,55 +85,55 @@ class MemoryController extends DisposableController
     @visibleForTesting DiffPaneController? connectedDiff,
     @visibleForTesting ProfilePaneController? connectedProfile,
   ) async {
-    assert(!_initialized.isCompleted);
+    assert(!_dataInitialized.isCompleted);
     switch (_mode) {
       case DevToolsMode.disconnected:
         // TODO(polina-c): load memory screen in disconnected mode, https://github.com/flutter/devtools/issues/6972
-        _initializeData();
+        await _initializeData();
       case DevToolsMode.connected:
-        _initializeData(
+        await _initializeData(
           diffPaneController: connectedDiff,
           profilePaneController: connectedProfile,
         );
       case DevToolsMode.offlineData:
         assert(connectedDiff == null && connectedProfile == null);
-        await maybeLoadOfflineData(
+        final loaded = await maybeLoadOfflineData(
           ScreenMetaData.memory.id,
           createData: (json) {
             final data = json[_jsonKey];
             if (data is OfflineMemoryData) return data;
             return OfflineMemoryData.fromJson(data as Map<String, dynamic>);
           },
           shouldLoad: (data) => true,
-          loadData: (data) => _initializeData(offlineData: data),
+          loadData: (data) async => await _initializeData(offlineData: data),
         );
         // [maybeLoadOfflineData] will be a noop if there is no offline data for the memory screen,
         //  so ensure we still call [_initializedData] if it has not been called.
-        if (!_initialized.isCompleted) _initializeData();
-        assert(_initialized.isCompleted);
+        if (!loaded) {
```

Comment:
it seems `loaded` is already local here....

Reply:
The `loaded` local variable was added in this PR on line 100. I'm saying that I think we can remove the `loaded` var (i.e. call `await maybeLoadOfflineData` without saving the result), and then revert this line back to `if (!_initialized.isCompleted) _initializeData();` without checking `if (loaded)`.

Reply:
Yes, it will be equivalent. Do you think removing `loaded` will improve readability?

Reply:
Checking `loaded` seems like we are adding another variable to check initialized state, when really, we want to gate calling `_initializeData` on whether or not `_dataInitialized` is completed. Adding another variable adds the risk that `loaded` and `_dataInitialized` will get out of sync. So yes, I think it is best that we check the completer directly, especially since `_initializeData` will throw an exception if the completer is already completed.

Reply:
+1, it's better to just use `!_initialized.isCompleted` and leave a comment if we're concerned about it not being obvious what the condition is meant to check.

Reply:
How about different perspective: we assume that _initializeData will not be completed not completed if data are not loaded by maybeLoadOfflineData. 
By adding variable `loaded` we verify this assumption. Without it we leave risk that data are actually loaded but _dataInitialized is not completed because of an error. Or another one: data are not loaded but _dataInitialized is completed.

In short: yes we have this risk to get out of sync, but with `loaded` we will catch it, and without it we will miss it.

Does it change something?

Reply:
Added assert as discussed with Kenzie.